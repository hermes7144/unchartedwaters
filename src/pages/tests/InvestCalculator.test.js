import { screen, render } from '@testing-library/react';
import { Route } from 'react-router-dom';
import { withInvest, withRouter } from '../../tests/utils';
import renderer from 'react-test-renderer'
import InvestCalculator from '../../components/InvestCalculator';

describe('InvestCalculator', () => {
  afterEach(() => {

  })
  it('renders correctly', () => {
    const component = renderer.create(
      withInvest(
        withRouter(<Route path='/' element={<InvestCalculator />} />)
      )
    );
    expect(component.toJSON()).toMatchSnapshot();
  })
})