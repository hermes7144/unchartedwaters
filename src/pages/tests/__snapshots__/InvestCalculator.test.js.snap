// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestCalculator renders correctly 1`] = `
<article
  className="w-full basis-1/2"
>
  <h2
    className="text-2xl font-bold my-4 text-center"
  >
    Calculator
  </h2>
  <form
    className="flex flex-col px-12"
  >
    <label
      htmlFor="target"
    >
      목표값
    </label>
    <input
      aria-label="target"
      id="target"
      maxLength="4"
      name="target"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="current"
    >
      현재값
    </label>
    <input
      aria-label="current"
      id="current"
      maxLength="4"
      name="current"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label>
      필요금액(독점)
    </label>
    <input
      aria-label="moneyMono"
      disabled={true}
      value=""
    />
    <label>
      필요금액(비독점)
    </label>
    <input
      aria-label="moneyNonMono"
      disabled={true}
      value=""
    />
    <label>
      투자횟수(독점)
    </label>
    <input
      aria-label="countMono"
      disabled={true}
      value=""
    />
    <label>
      투자횟수(비독점)
    </label>
    <input
      aria-label="countNonMono"
      disabled={true}
      value=""
    />
    <label>
      라파엘 군사투자(독점)
    </label>
    <input
      aria-label="rapaelMono"
      disabled={true}
      value=""
    />
    <label>
      라파엘 군사투자(비독점)
    </label>
    <input
      aria-label="rapaelNonMono"
      disabled={true}
      value=""
    />
  </form>
</article>
`;
